sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -66,9 +65,12 @@ public static MetricStorage empty() {
         new ObservableDoubleMeasurement() {
           @Override
           public void observe(double value, Attributes attributes) {
-            measurementAccumulator.record(
-                attributesProcessor.process(attributes, Context.current()),
-                aggregator.accumulateDoubleMeasurement(value, attributes, Context.current()));
+            T accumulation =
+                aggregator.accumulateDoubleMeasurement(value, attributes, Context.current());
+            if (accumulation != null) {
```

Comment:
Accumulation should not be `null`, and we should enforce that.

Reply:
Would you be able to follow up after merging this? Given the return value is an arbitrary `T`, I don't see any way to easily fix this line to not return null, which is what's propagating up to here.

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/aggregator/AggregatorHandle.java#L94

Reply:
We use `null` to denote no-recording and drop bad cells in the Map of Attribute -> value.  However that's only for Synchronous storage, I'm curious how it's bubbling up to here, how can I look into that?


The TL;DR: though is you should consider an accumulation of `null` here a bug, and we should throw/log so we can fix the logic error.

Reply:
I think this is the same as @jack-berg's proposal above. I'd like to merge this before making such a large refactoring